src/engine/strat_engine/util.rs

```diff
@@ -72,3 +79,32 @@ pub fn set_uuid(devnode: &Path, uuid: Uuid) -> EngineResult<()> {
         Err(EngineError::Engine(ErrorEnum::Error, err_msg))
     }
 }
+
+/// Lookup the WWN from the udev db using the device node eg. /dev/sda
+pub fn hw_lookup(dev_node_search: &Path) -> strat_engine_util::Result<Option<String>> {
+    let context = libudev::Context::new()
+        .chain_err(|| "Unable to construct libudev context")?;
+    let mut enumerator = libudev::Enumerator::new(&context)
+        .chain_err(|| "Unable to construct enumerator")?;
+    enumerator
+        .match_subsystem("block")
+        .chain_err(|| "Unable to match udev subsystem")?;
+    enumerator
+        .match_property("DEVTYPE", "disk")
+        .chain_err(|| "Unable to match udev property")?;
+
+    for device in enumerator
+            .scan_devices()
+            .chain_err(|| "Unable to scan udev db")? {
+        if dev_node_search == device.devnode().chain_err(|| "device.devnode error")? {
+            if let Some(wwid) = device.property_value("ID_WWN") {
+                return Ok(Some(String::from(wwid.to_str()
```

Comment:
I don't know why you need ```to_str``` here.

Reply:
The property value is returned as an `OsString` just trying to get it into a `String` format.